include/seqan3/std/ranges

```diff
@@ -173,8 +179,171 @@ template <typename type>
 
+/*!\interface std::ranges::semiregular_t
+* \brief Alias for ranges::semiregular_t. Obtains the value type of a Semiregular type.
+*/
+using ::ranges::semiregular_t;
+
+/*!\interface std::ranges::common_reference_t
+* \brief Alias for ranges::common_reference_t. Determine the common reference type of a set of types.
+*/
+using ::ranges::common_reference_t;
+
+/*!\interface std::ranges::common_type_t
+* \brief Alias for ranges::common_type_t. Determine the common type of a set of types.
+*/
+using ::ranges::common_type_t;
+
+/*!\interface std::ranges::uncvref_t
+* \brief Alias for ranges::uncvref_t. Obtains the reference type of a top level cv.
+*/
+using ::ranges::uncvref_t;
+
+/*!\interface std::ranges::range_value_type_t
+* \brief Alias for ranges::range_value_type_t. Obtains the value type of a Range type.
+*/
+using ::ranges::range_value_type_t;
+
+/*!\interface std::ranges::ostreambuf_iterator
+* \brief Alias for ranges::ostreambuf_iterator. Output iterator that writes to std::basic_streambuf.
+*/
+using ::ranges::ostreambuf_iterator;
+
+/*!\interface std::ranges::back_insert_iterator
+* \brief Alias for ranges::back_insert_iterator. Iterator adaptor for insertion at the end of a container.
+*/
+using ::ranges::back_insert_iterator;
+
+/*!\interface std::ranges::iterator_range
+* \brief Alias for ranges::iterator_range. Iterator adaptor for a Range type.
+*/
+using ::ranges::iterator_range;
+
+/*!\interface std::ranges::make_iterator_range
+* \brief Alias for ranges::make_iterator_range. Makes the iterator adaptor.
+*/
+using ::ranges::make_iterator_range;
+
+/*!\interface std::ranges::back_inserter
+* \brief Alias for ranges::back_inserter. Creates the back inserter.
+*/
+using ::ranges::back_inserter;
+
+/*!\interface std::ranges::cbegin
+* \brief Alias for ranges::cbegin. Returns an iterator to the beginning of a range.
+*/
+using ::ranges::cbegin;
+
+/*!\interface std::ranges::cend
+* \brief Alias for ranges::cend. Returns an iterator to the end of a range.
+*/
+using ::ranges::cend;
+
+/*!\interface std::ranges::istream_range
+* \brief Alias for ranges::istream_range. Istream adaptor for a Range type.
+*/
+using ::ranges::istream_range;
+
+/*!\interface std::ranges::ostream_iterator
+* \brief Alias for ranges::ostream_iterator. Ostream adaptor for a Range type.
+*/
+using ::ranges::ostream_iterator;
+
+/*!\interface std::ranges::zip_view
+* \brief Alias for ranges::zip_view.
+*/
+using ::ranges::zip_view;
+
+/*!\interface std::ranges::any_view
+* \brief Alias for ranges::any_view.
+*/
+using ::ranges::any_view;
+
+/*!\interface std::ranges::reverse_view
+* \brief Alias for ranges::reverse_view.
+*/
+using ::ranges::reverse_view;
+
+/*!\interface std::ranges::ImplicitlyConvertibleTo
+* \brief Alias for ranges::ImplicitlyConvertibleTo.
+*/
+using ::ranges::ImplicitlyConvertibleTo;
+
+/*!\interface std::ranges::ExplicitlyConvertibleTo
+* \brief Alias for ranges::ExplicitlyConvertibleTo.
+*/
+using ::ranges::ExplicitlyConvertibleTo;
```

Comment:
don't add these two, instead use our concepts from core/concept

Reply:
But the implementation of `explicitly_convertible_to_concept` or `implicitly_convertible_to_concept` still rely on `std::ranges::ExplicitlyConvertibleTo`, right? The implementation in core/concept is used throughout seqan, but in core/concept/core_language.hpp, I need to use `std::ranges::ExplicitlyConvertibleTo` or something similar.

Reply:
good point. You can change the definition of implicitly_convertible_concept to delegate to `std::is_convertible_v` and for the explicitly_convertible_to_concept check if a `static_cast`  would work. @smehringer can help you with this if something does not work.

Reply:
Should it be `std::is_convertible_v` or `std::is_nothrow_convertible_v`? (https://en.cppreference.com/w/cpp/types/is_convertible)

Reply:
Use 
```
template <typename t, typename u>
concept explicitly_convertible_to_concept = requires (t vt) { {static_cast<u>(vt)}; };
```